docarray/array/storage/weaviate/backend.py

```diff
@@ -0,0 +1,290 @@
+from dataclasses import dataclass, field
+import uuid
+import itertools
+from typing import (
+    Generator,
+    Iterator,
+    Dict,
+    Sequence,
+    Optional,
+    TYPE_CHECKING,
+    Union,
+    Tuple,
+    List,
+)
+
+import weaviate
+import uuid
+import scipy.sparse
+
+from .... import Document
+from ..base.backend import BaseBackendMixin
+
+if TYPE_CHECKING:
+    from ....types import (
+        DocumentArraySourceType,
+    )
+
+
+@dataclass
+class WeaviateConfig:
+    """This class stores the config variables to initialize
+    connection to the Weaviate server"""
+
+    client: Optional[Union[str, weaviate.Client]] = None
+    n_dim: Optional[int] = None
+    array_id: Optional[str] = None
```

Comment:
`array_id` is not an explicit name to tell the user what's the meaning? And how to use it. How about renaming as `name`

Reply:
Sounds good, I will rename it.